sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -108,25 +97,33 @@ private String ensureIndexed(List<Object> fieldNames,
             fieldNamesList.add(0, "_id");
         }
 
-        // Does the index already exist; return success if it does and is same, else fail
+        // Check the index limit.  Limit is 1 for "text" indexes and unlimited for "json" indexes.
+        // Then check whether the index already exists; return success if it does and is same,
+        // else fail.
         try {
             Map<String, Object> existingIndexes = listIndexesInDatabaseQueue();
-            if (existingIndexes != null && existingIndexes.get(indexName) != null) {
-                @SuppressWarnings("unchecked")
-                Map<String, Object> index = (Map<String, Object>) existingIndexes.get(indexName);
-                String existingType = (String) index.get("type");
-                @SuppressWarnings("unchecked")
-                List<String> existingFieldsList = (List<String>) index.get("fields");
+            if (indexLimitReached(index, existingIndexes)) {
+                String msg = String.format("Index limit reached.  Cannot create index %s.",
+                                           index.indexName);
```

Comment:
Either here or in `indexLimitReached` the log message needs to indicate what "index limit reached" means. Nowhere do you say something like "You already have a text index, %s. You can only define a single text index per datastore. Delete %s and recreate to include the new fields.".